docs/logs.md

```diff
@@ -0,0 +1,221 @@
+      tag_on_failure => ["_routerparsefailure"]
+      add_tag => ["gorouter"]
+    }
+    # replace @timestamp field with the one from router access log
+    date {
+      match => [ "router_timestamp", "ISO8601" ]
+    }
+    kv {
+      source => "router_keys"
+      target => "router"
+      value_split => ":"
+      remove_field => "router_keys"
+    }
+  }
+
+  # Application logs
+  if [syslog_proc] =~ "APP\/PROC\/WEB" {
+    grok {
+      match => { "syslog_msg" => "NginxLog \"%{WORD:[access][method]} %{NOTSPACE:[access][url]} HTTP/%{NUMBER:[access][http_version]}\" %{NONNEGINT:[access][response_code]:int} %{NONNEGINT:[access][body_received][bytes]:int}" }
+      tag_on_failure => ["_appparsefailure"]
+      add_tag => ["app"]
+    }
+  }
+
+  # Extract file extension from URL
+  if [access][url] {
+      grok {
+          match => { "[access][url]" => ".+\.%{WORD:[access][extension]}" }
+      }
+  }
+
+  # Extract query string
+  if [access][url] {
+    grok {
+      match => [ "[access][url]", "%{URIPARAM:[access][querystring]}" ]
+    }
+  }
+
+  # Anonymise IP addresses
+  # https://www.elastic.co/guide/en/logstash/current/plugins-filters-fingerprint.html
+  # Fingerprinting the IP address removes the ‘host part’, which makes the IPs less likely to identify someone. (key: 16)
+  if [access][user_ip] {
+    fingerprint {
+      key => "16"
+      source => "[access][user_ip]"
+      target => "[access][user_ip]"
+      method => "IPV4_NETWORK"
+    }
+    # Anonymised IP is passed to Geo IP to give high level region information, for example to find people from the UK
+    # https://www.elastic.co/guide/en/logstash/current/plugins-filters-geoip.html
+    geoip { source => "[access][user_ip]" }
+  }
+
+  # Replace the message and source_host fields
+  mutate {
+    rename => [ "syslog_host", "source_host" ]
+    rename => [ "syslog_msg", "message" ]
+  }
+
+  # Update the original message to remove potentially personally identifable information
+  mutate {
+    gsub => [
+      # redact all IP addresses
+      # You can't use grok patterns so copying manually the IPv4 pattern from https://github.com/logstash-plugins/logstash-patterns-core/blob/master/patterns/grok-patterns
+      "message", "(?<![0-9])(?:(?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])[.](?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])[.](?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])[.](?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5]))(?![0-9])", "[redacted-ip]"
+    ]
+    tag_on_failure => ["_redactingmessagefailure"]
+  }
+}
+```
+
+## Filter plugins
```

Comment:
Should this section be placed further down in the file? I think the below section `### How the filter turns the raw logs into fields` is a subsection of `## How we transform logs with Logstash Filters`